src/Our.Umbraco.Ditto/ComponentModel/Processors/DittoProcessorRegistry.cs

```diff
@@ -31,6 +31,17 @@ internal class DittoProcessorRegistry
         private Type DefaultProcessorType = typeof(UmbracoPropertyAttribute);
 
         /// <summary>
+        /// The default post-processors.
+        /// </summary>
+        private List<DittoProcessorAttribute> DefaultPostProcessorAttributes = new List<DittoProcessorAttribute>()
```

Comment:
I know nothing about thread safety, but given we protect the Cache collection, I wonder if we need to do the something with the list. I'm not sure if there is anything baked in we can used, or whether we'd need to use another lock, but worth thinking about.

Reply:
OK cool, I'll add a lock in.  I'm wondering if we can reuse `CacheLock` for it?

...and for when adding/inserting, do you think we should check if the processor type has already been added? (prevent duplicates)

Reply:
I don't think  you should use the same lock as you could end up holding up the other function, so probably go with another lock.

I'm wondering if there is ever the need to have the same processor more than once? Could be possible, but can't think of a use case atm. Happy either way.